builtin/game/knockback.lua

```diff
@@ -22,14 +22,19 @@ local function vector_absmax(v)
 	return max(max(abs(v.x), abs(v.y)), abs(v.z))
 end
 
-core.register_on_punchplayer(function(player, hitter, time_from_last_punch, tool_capabilities, unused_dir, damage)
+core.register_on_punchplayer(function(player, hitter, time_from_last_punch, tool_capabilities, dir, damage)
+	if not hitter and not dir then
```

Comment:
I think you don't need to check for `not dir` because `dir = nil` will default to the zero vector anyways (which works fine for our purposes and should be documented)

Reply:
@sfence

Reply:
Removed the check entirely, it is unnecessary: `dir` will default to the zero vector as it should (tested this as well just to be sure), so this will never trigger. `hitter` being `nil` is guarded against in the following code as it should, while still supporting `nil` hitter knockback properly. This is passed on to `minetest.calculate_knockback`, which refers to the now updated on punchplayer documentation. So we're good now.